src/data/java_profile.rs

```diff
@@ -0,0 +1,340 @@
+
+pub static JAVA_PROFILE_FILE_NAME: &str = "java_profile";
+
+lazy_static! {
+    pub static ref ASPROF_CHILDREN: Mutex<Vec<Child>> = Mutex::new(Vec::new());
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct JavaProfileRaw {
+    process_map: HashMap<String, Vec<String>>,
+}
+
+impl JavaProfileRaw {
+    pub fn new() -> Self {
+        JavaProfileRaw {
+            process_map: HashMap::new(),
+        }
+    }
+
+    fn launch_asprof(&self, jids: Vec<String>, params: CollectorParams) -> Result<()> {
+        let data_dir = PathBuf::from(params.data_dir.clone());
+        for jid in &jids {
+            let mut html_loc = data_dir.clone();
+            html_loc.push(format!("{}-java-flamegraph-{}.html", params.run_name, jid));
+
+            match Command::new("asprof")
+                .args([
+                    "-d",
+                    &(params.collection_time - params.elapsed_time).to_string(),
+                    "-f",
+                    html_loc.to_str().unwrap(),
+                    jid.as_str(),
+                ])
+                .spawn()
+            {
+                Err(e) => {
+                    if e.kind() == ErrorKind::NotFound {
+                        error!("'asprof' command not found.");
+                    } else {
+                        error!("Unknown error: {}", e);
+                    }
+                    error!("Skipping asprof profile collection.");
+                }
+                Ok(child) => {
+                    debug!(
+                        "Recording asprof profiling data for '{}' with PID, {}.",
+                        self.process_map
+                            .get(jid.as_str())
+                            .unwrap_or(&vec![String::from("JVM")])[0],
+                        jid
+                    );
+                    ASPROF_CHILDREN.lock().unwrap().push(child);
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn get_jid(&mut self, key: &str) -> Option<String> {
+        if self.process_map.contains_key(key) {
+            return Some(key.to_string());
+        }
+
+        self.process_map.iter().find_map(|(jid, name)| {
+            if key == name[0] {
+                Some(jid.to_string())
+            } else {
+                None
+            }
+        })
+    }
+
+    fn update_process_map(&mut self) -> String {
+        debug!("Running jps (may incur utilization spike)...");
+        let jps_cmd = Command::new("jps").output();
+        let mut jps_str = String::new();
+        match jps_cmd {
+            Ok(jps_out) => {
+                jps_str = String::from_utf8(jps_out.stdout).unwrap_or_default();
+                let jps: Vec<&str> = jps_str.split_whitespace().collect();
+                for i in (0..jps.len()).step_by(2) {
+                    if jps[i + 1] != "Jps" {
+                        self.process_map
+                            .insert(String::from(jps[i]), vec![String::from(jps[i + 1])]);
```

Comment:
In code that's parsing little strings like this, it's helpful to add a comment with an example, so readers can have a sense for the values being dealt with.